galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+                         Note that screen_size may be tweaked by the initializer to ensure
+                         screen_size is a multiple of screen_scale.
+    @param screen_scale  Physical pixel scale of phase screen in meters.  A fraction of the Fried
+                         parameter is usually sufficiently small, but users should test the effects
+                         of this parameter to ensure robust results. [Default: half of r0_500]
+    @param altitude      Altitude of phase screen in km.  This is with respect to the telescope, not
+                         sea-level.  [Default: 0.0]
+    @param time_step     Interval to use when advancing the screen in time in seconds.
+                         [Default: 0.03]
+    @param r0_500        Fried parameter setting the amplitude of turbulence; contributes to "size"
+                         of the resulting atmospheric PSF.  Specified at wavelength 500 nm, in units
+                         of meters.  [Default: 0.2]
+    @param L0_inv        Inverse outer scale in inverse meters, beyond which turbulence power
+                         spectrum smoothly approaches zero.  [Default: 1./25]
+    @param vx            x-component wind velocity in meters/second.  [Default: 0.]
+    @param vy            y-component wind velocity in meters/second.  [Default: 0.]
+    @param rng           Random number generator as a galsim.BaseDeviate().  If None, then use the
+                         clock time or system entropy to seed a new generator.  [Default: None]
+    """
+    def __init__(self, screen_size, screen_scale=None, altitude=0.0, time_step=0.03,
+                 r0_500=0.2, L0_inv=1./25.0, vx=0.0, vy=0.0, rng=None):
+        super(FrozenAtmosphericScreen, self).__init__(screen_size, screen_scale, altitude,
+                                                      time_step, r0_500, L0_inv,
+                                                      vx, vy, rng)
+
+        # Note that _init_screen is here instead of in superclass since ARAtmosphericScreen reuses
+        # fx, fy, but we don't want to store these in the class or recompute them.  So instead, we
+        # just provide the `_freq` method in the superclass that each AtmosphericScreen subclass can
+        # use.
+        fx, fy = self._freq()
+        self._init_screen(fx, fy)
+        # Use a LookupTable2D to interpolate/extrapolate with periodic boundary conditions.
+        nx = self.screen_size/self.screen_scale
+        # x0 = y0 = -self.screen_size/2.0
+        x0 = y0 = -0.5*(nx-1)*self.screen_scale
+        dx = dy = self.screen_scale
+
+        self.tab2d = galsim.LookupTable2D(x0, y0, dx, dy, self.screen, edge_mode='wrap')
+        # Don't need these anymore, so free memory
+        del self._phaseFT, self.psi, self.screen
+
+        # To handle wind, we will interpolate the LookupTable2D with an offset origin.
+        self.origin = np.r_[0.0, 0.0]
```

Comment:
And here, I'd use a call to `self.reset()` to encapsulate what needs to happen at the start in one place.